components/UserProfilePage.js

```diff
@@ -3,18 +3,24 @@ const Component = require('./Component')
 const Comment = require('./Comment')
 
 module.exports = class UserProfilePage extends Component {
-  render() {
-    const { proxied_name, selected_profile: p, user } = this.state
+  oninit() {
+    if (!this.state.user) return this.location.redirect('/sign_in')
```

Comment:
This shouldn't be here. This redirects to the sign-in page, but user profiles should be visible to anyone.